pkg/command/context_test.go

```diff
@@ -830,6 +894,313 @@ var _ = Describe("testing context use", func() {
 	})
 })
 
+func Test_completionContext(t *testing.T) {
+	ctxK8s1 := &configtypes.Context{
+		Name:        "tkg1",
+		Target:      configtypes.TargetK8s,
+		ClusterOpts: &configtypes.ClusterServer{Endpoint: "https://example.com/myendpoint/k8s/1"},
+	}
+	ctxK8s2 := &configtypes.Context{
+		Name:        "tkg2",
+		Target:      configtypes.TargetK8s,
+		ClusterOpts: &configtypes.ClusterServer{Path: "/example.com/mypath/k8s/2", Context: "ctxTkg2"},
+	}
+	ctxTMC1 := &configtypes.Context{
+		Name:       "tmc1",
+		Target:     configtypes.TargetTMC,
+		GlobalOpts: &configtypes.GlobalServer{Endpoint: "https://example.com/myendpoint/tmc/1"},
+	}
+	ctxTMC2 := &configtypes.Context{
+		Name:       "tmc2",
+		Target:     configtypes.TargetTMC,
+		GlobalOpts: &configtypes.GlobalServer{Endpoint: "https://example.com/myendpoint/tmc/2"},
+	}
+	ctxTAE1 := &configtypes.Context{
+		Name:        "tae1",
+		Target:      configtypes.TargetTAE,
+		ClusterOpts: &configtypes.ClusterServer{Endpoint: "https://example.com/myendpoint/tae/1"},
+	}
+	ctxTAE2 := &configtypes.Context{
+		Name:        "tae2",
+		Target:      configtypes.TargetTAE,
+		ClusterOpts: &configtypes.ClusterServer{Endpoint: "https://example.com/myendpoint/tae/2"},
+	}
+
+	expectedOutForAllCtxs := ctxTAE1.Name + "\t" + ctxTAE1.ClusterOpts.Endpoint + "\n"
+	expectedOutForAllCtxs += ctxTAE2.Name + "\t" + ctxTAE2.ClusterOpts.Endpoint + "\n"
+	expectedOutForAllCtxs += ctxK8s1.Name + "\t" + ctxK8s1.ClusterOpts.Endpoint + "\n"
+	expectedOutForAllCtxs += ctxK8s2.Name + "\t" + ctxK8s2.ClusterOpts.Path + ":" + ctxK8s2.ClusterOpts.Context + "\n"
+	expectedOutForAllCtxs += ctxTMC1.Name + "\t" + ctxTMC1.GlobalOpts.Endpoint + "\n"
+	expectedOutForAllCtxs += ctxTMC2.Name + "\t" + ctxTMC2.GlobalOpts.Endpoint + "\n"
+
+	expectedOutForActiveCtxs := ctxTAE1.Name + "\t" + ctxTAE1.ClusterOpts.Endpoint + "\n"
+	expectedOutForActiveCtxs += ctxK8s1.Name + "\t" + ctxK8s1.ClusterOpts.Endpoint + "\n"
+	expectedOutForActiveCtxs += ctxTMC1.Name + "\t" + ctxTMC1.GlobalOpts.Endpoint + "\n"
+
+	expectedOutForTMCActiveCtx := ctxTMC1.Name + "\t" + ctxTMC1.GlobalOpts.Endpoint + "\n"
+
+	expectedOutForTAECtxs := ctxTAE1.Name + "\t" + ctxTAE1.ClusterOpts.Endpoint + "\n"
+	expectedOutForTAECtxs += ctxTAE2.Name + "\t" + ctxTAE2.ClusterOpts.Endpoint + "\n"
+
+	expectedOutforTargetFlag := compK8sTarget + "\n" + compTAETarget + "\n" + compTMCTarget + "\n"
+
+	kubeconfigFile1, err := os.CreateTemp("", "kubeconfig")
```

Comment:
nit: missing cleanup of the kubeconfig files created

Reply:
Good eye!